src/odoc/bin/main.ml

```diff
@@ -187,14 +192,34 @@ end = struct
               "Either --package or --parent should be specified, not both")
     in
     let source =
-      match (source_parent_file, source_name) with
-      | Some parent, Some name -> Ok (Some (parent, name, source_cmt))
-      | Some _, None | None, Some _ ->
+      match
+        (source_parent_file, source_name, source_cmt, Fs.File.get_ext input)
+      with
+      | Some parent, Some name, None, ".cmt" -> Ok (Some (parent, name, input))
+      | Some parent, Some name, Some cmt, ".cmt" ->
+          if Fpath.equal cmt input then Ok (Some (parent, name, input))
+          else
+            Error
+              (`Cli_error
+                "--cmt has to be equal to the input file when this one has \
+                 .cmt extension.")
```

Comment:
This seems redundant - why don't we always read the shape if we're handed a cmt file?

Reply:
If we allow for a `--cmt` argument for compiling `.cmti` files, then the following command can be used: `odoc compile --cmt <file1.cmt> <file2.cmt>`.

There are several reactions we can have:
1. Forbid the `--cmt` argument when the input is a `.cmt` file, and use the input.
1. Only accept that if the input is consistent: either there is no`--cmt` argument, or `--cmt` and input are the same.
1. Make `--cmt` have precedence over the input file

I decided to go for 2. But yes, we don't need the `--cmt` argument if we are handed a cmt file.